### PR TITLE
Move span management to servlet

### PR DESCRIPTION
**Pull request by wendigo**
The additional information extracted from JAX-RS is used to update span

# Airlift contribution check list

 - [x] Git commit messages follow https://cbea.ms/git-commit/.
 - [x] All automated tests are passing.
 - [x] Pull request was categorized using one of the existing labels.

### CODE REVIEW

**Review by oskar-szwajkowski — APPROVED** [review]
LGTM, should provide handling of cases mentioned in: https://github.com/airlift/airlift/pull/1328 due to moving filter from REST-layer to http-layer

**Comment by wendigo** [comment]
Screenshot 2024-12-03 at 15 09 40

File: jaxrs/src/main/java/io/airlift/jaxrs/tracing/TracingServletFilter.java [hunk 1]
```diff
@@ -73,6 +74,13 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                 .setAttribute(ClientAttributes.CLIENT_ADDRESS, request.getRemoteAddr())
                 .setAttribute(NetworkAttributes.NETWORK_PROTOCOL_NAME, "http");
 
+        // https://github.com/jetty/jetty.project/blob/jetty-12.0.x/jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextRequest.java#L63
+        String sessionId = (String) request.getAttribute("jakarta.servlet.request.ssl_session_id");
```
Comment (martint) [inline jaxrs/src/main/java/io/airlift/jaxrs/tracing/TracingServletFilter.java:78]: Is this guaranteed, or an implementation detail of Jetty?

File: jaxrs/src/main/java/io/airlift/jaxrs/tracing/TracingServletFilter.java [hunk 2]
```diff
@@ -1,44 +1,152 @@
 package io.airlift.jaxrs.tracing;
```
Comment (martint) [inline jaxrs/src/main/java/io/airlift/jaxrs/tracing/TracingServletFilter.java:1]: This is the wrong package/module. It should go in the HTTP server module.

File: jaxrs/src/main/java/io/airlift/jaxrs/tracing/TracingServletFilter.java (hunk 1 above)
Comment (wendigo) [reply jaxrs/src/main/java/io/airlift/jaxrs/tracing/TracingServletFilter.java:78]: It works in Tomcat, Jetty, Undertow.

File: jaxrs/src/main/java/io/airlift/jaxrs/tracing/TracingServletFilter.java (hunk 2 above)
Comment (wendigo) [reply jaxrs/src/main/java/io/airlift/jaxrs/tracing/TracingServletFilter.java:1]: sure, I can move it

Comment (wendigo) [reply jaxrs/src/main/java/io/airlift/jaxrs/tracing/TracingServletFilter.java]: Moved.

File: jaxrs/src/main/java/io/airlift/jaxrs/tracing/TracingServletFilter.java (hunk 1 above)
Comment (wendigo) [reply jaxrs/src/main/java/io/airlift/jaxrs/tracing/TracingServletFilter.java]: When moved in the last commit, I can now reference the public static const from Jetty directly.

**Comment by wendigo** [comment]
@martint comments addressed

**Comment by wendigo** [comment]
This is how it looks after this change:

Screenshot 2024-12-04 at 17 52 13

**Review by martint — APPROVED** [review]

**Pull request merged**